### ISSUE TITLE
JSON for Modern C++ seems to be installed (header json.hpp)

### ISSUE THREAD

**Issue by nlohmann** [melpon/wandbox#209]
I am the author of [JSON for Modern C++](https://github.com/nlohmann/json) and I heavily use Wandbox for my documentation. Thanks for the great service!!!

However, in the last weeks I have problems compiling an example, see https://wandbox.org/permlink/GkVyS3dqnUwGeUqO. The uploaded header file contains the function that the compiler could not find - I can compile the code without problems on my machine.

Then, I found out that the following code compiles

```cpp
#include <iostream>
#include <json.hpp>

using json = nlohmann::json;

int main()
{
    std::cout << json::meta() << std::endl;
}
```

with any C++11 compiler **even if there is no file "json.hpp" uploaded**, see screenshot:
 
![wandbox](https://cloud.githubusercontent.com/assets/159488/25048348/6471da9a-213c-11e7-97d8-2411e685559f.png)

It seems as if the header `json.hpp` somehow got copied to a place where it is reachable from the compiler.

-----

### Steps to reproduce

- open https://wandbox.org/
- paste above code
- click "run"

### Observed behavior

Output

```
{"compiler":{"c++":"201703","family":"gcc","version":"7.0.1"},"copyright":"(C) 2013-2017 Niels Lohmann","name":"JSON for Modern C++","platform":"linux","url":"https://github.com/nlohmann/json","version":{"major":2,"minor":1,"patch":1,"string":"2.1.1"}}
```

### Expected behavior

```
prog.cc:2:10: fatal error: json.hpp: No such file or directory
 #include <json.hpp>
          ^~~~~~~~~~~
compilation terminated.
```

**Comment by melpon** [comment]
Thank you for using Wandbox!!!

Yes, your JSON library is installed to Wandbox since [this commit](https://github.com/melpon/wandbox-builder/commit/cfaf452e1fefc3b6d5a628cef89b629ec1d93f2a).
Wandbox C++ compiler specifies an include path of the library explicitly.
The library is updated every day.

So, you may not copy `json.hpp` to Wandbox if you use the HEAD version library.
If you copy `json.hpp` to Wandbox, use `#include "json.hpp"` instead of `#include <json.hpp>`.

**Comment by nlohmann** [comment]
Thanks a lot! I fixed my examples with https://github.com/nlohmann/json/commit/9b32f7258421cf747d0984ed28d273530e6f8587.

**Comment by NightWarGames** [comment]
I am also having trouble using WandBox with JSON. When I compile I get the error:

```
#include <iostream>
#include <cstdlib>
#include <json.hpp>

int main()
{
    std::cout << "Hello, Wandbox!" << std::endl;
}


//////// OUTPUT
prog.cc:5:10: fatal error: 'json.hpp' file not found
#include <json.hpp>
         ^~~~~~~~~~
1 error generated.
```

I have tried about 5 different versions of compilers from GCC 8.1 to HEAD and some Clang. I have tried with C++11-C++17 and GNU versions. The file isn't there, and when trying to add the file manually it gets many errors:
```

In file included from /opt/wandbox/gcc-head/include/c++/10.0.0/bits/move.h:55,
                 from /opt/wandbox/gcc-head/include/c++/10.0.0/bits/nested_exception.h:40,
                 from /opt/wandbox/gcc-head/include/c++/10.0.0/exception:148,
                 from /opt/wandbox/gcc-head/include/c++/10.0.0/ios:39,
                 from /opt/wandbox/gcc-head/include/c++/10.0.0/ostream:38,
                 from /opt/wandbox/gcc-head/include/c++/10.0.0/iostream:39,
                 from prog.cc:2:
/opt/wandbox/gcc-head/include/c++/10.0.0/type_traits: In instantiation of 'struct std::is_constructible<nlohmann::basic_json<>, const nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer>&>':
json.hpp:11017:79:   required by substitution of 'template<class ... Args, typename std::enable_if<std::is_constructible<nlohmann::basic_json<>, Args ...>::value, int>::type <anonymous> > nlohmann::detail::json_ref<nlohmann::basic_json<> >::json_ref(Args&& ...) [with Args = {const nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer>&}; typename std::enable_if<std::is_constructible<nlohmann::basic_json<>, Args ...>::value, int>::type <anonymous> = <missing>]'
/opt/wandbox/gcc-head/include/c++/10.0.0/type_traits:895:30:   required from 'struct std::__is_constructible_impl<nlohmann::basic_json<>, const nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer>&>'
/opt/wandbox/gcc-head/include/c++/10.0.0/type_traits:924:12:   required from 'struct std::__is_copy_constructible_impl<nlohmann::basic_json<>, true>'
/opt/wandbox/gcc-head/include/c++/10.0.0/type_traits:930:12:   required from 'struct std::is_copy_constructible<nlohmann::basic_json<> >'
/opt/wandbox/gcc-head/include/c++/10.0.0/type_traits:138:12:   [ skipping 2 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/opt/wandbox/gcc-head/include/c++/10.0.0/type_traits:2651:31:   required by substitution of 'template<class _Tp, class> static std::true_type std::__swappable_details::__do_is_swappable_impl::__test(int) [with _Tp = nlohmann::basic_json<>; <template-parameter-1-2> = <missing>]'
/opt/wandbox/gcc-head/include/c++/10.0.0/type_traits:2675:35:   required from 'struct std::__is_swappable_impl<nlohmann::basic_json<> >'
/opt/wandbox/gcc-head/include/c++/10.0.0/type_traits:2701:12:   required from 'struct std::is_swappable<nlohmann::basic_json<> >'
/opt/wandbox/gcc-head/include/c++/10.0.0/type_traits:2720:38:   required from 'constexpr const bool std::is_swappable_v<nlohmann::basic_json<> >'
/opt/wandbox/gcc-head/include/c++/10.0.0/optional:1175:51:   required by substitution of 'template<class _Tp> std::enable_if_t<(!(is_move_constructible_v<_Tp> && is_swappable_v<_Tp>))> std::swap(std::optional<_Tp>&, std::optional<_Tp>&) [with _Tp = nlohmann::basic_json<>]'
json.hpp:22527:1:   required from here
/opt/wandbox/gcc-head/include/c++/10.0.0/type_traits:900:12: error: invalid use of incomplete type 'struct std::__is_constructible_impl<nlohmann::basic_json<>, const nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer>&>'
  900 |     struct is_constructible
      |            ^~~~~~~~~~~~~~~~
/opt/wandbox/gcc-head/include/c++/10.0.0/type_traits:894:12: note: declaration of 'struct std::__is_constructible_impl<nlohmann::basic_json<>, const nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer>&>'
  894 |     struct __is_constructible_impl
      |            ^~~~~~~~~~~~~~~~~~~~~~~
In file included from /opt/wandbox/gcc-head/include/c++/10.0.0/vector:66,
                 from header.h:4,
                 from prog.cc:4:
/opt/wandbox/gcc-head/include/c++/10.0.0/bits/stl_uninitialized.h: In instantiation of '_ForwardIterator std::uninitialized_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = __gnu_cxx::__normal_iterator<const nlohmann::basic_json<>*, std::vector<nlohmann::basic_json<>, std::allocator<nlohmann::basic_json<> > > >; _ForwardIterator = nlohmann::basic_json<>*]':
/opt/wandbox/gcc-head/include/c++/10.0.0/bits/stl_uninitialized.h:323:37:   required from '_ForwardIterator std::__uninitialized_copy_a(_InputIterator, _InputIterator, _ForwardIterator, std::allocator<_Tp>&) [with _InputIterator = __gnu_cxx::__normal_iterator<const nlohmann::basic_json<>*, std::vector<nlohmann::basic_json<>, std::allocator<nlohmann::basic_json<> > > >; _ForwardIterator = nlohmann::basic_json<>*; _Tp = nlohmann::basic_json<>]'
/opt/wandbox/gcc-head/include/c++/10.0.0/bits/stl_vector.h:555:31:   required from 'std::vector<_Tp, _Alloc>::vector(const std::vector<_Tp, _Alloc>&) [with _Tp = nlohmann::basic_json<>; _Alloc = std::allocator<nlohmann::basic_json<> >]'
/opt/wandbox/gcc-head/include/c++/10.0.0/ext/new_allocator.h:152:4:   required from 'void __gnu_cxx::new_allocator<_Tp>::construct(_Up*, _Args&& ...) [with _Up = std::vector<nlohmann::basic_json<>, std::allocator<nlohmann::basic_json<> > >; _Args = {const std::vector<nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer>, std::allocator<nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer> > >&}; _Tp = std::vector<nlohmann::basic_json<>, std::allocator<nlohmann::basic_json<> > >]'
/opt/wandbox/gcc-head/include/c++/10.0.0/bits/alloc_traits.h:484:17:   required from 'static void std::allocator_traits<std::allocator<_CharT> >::construct(std::allocator_traits<std::allocator<_CharT> >::allocator_type&, _Up*, _Args&& ...) [with _Up = std::vector<nlohmann::basic_json<>, std::allocator<nlohmann::basic_json<> > >; _Args = {const std::vector<nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer>, std::allocator<nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer> > >&}; _Tp = std::vector<nlohmann::basic_json<>, std::allocator<nlohmann::basic_json<> > >; std::allocator_traits<std::allocator<_CharT> >::allocator_type = std::allocator<std::vector<nlohmann::basic_json<>, std::allocator<nlohmann::basic_json<> > > >]'
json.hpp:15278:35:   [ skipping 2 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
json.hpp:16215:25:   required from 'nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer>::basic_json(const nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer>&) [with ObjectType = std::map; ArrayType = std::vector; StringType = std::__cxx11::basic_string<char>; BooleanType = bool; NumberIntegerType = long int; NumberUnsignedType = long unsigned int; NumberFloatType = double; AllocatorType = std::allocator; JSONSerializer = nlohmann::adl_serializer]'
json.hpp:4634:31:   required from 'bool nlohmann::detail::json_sax_dom_callback_parser<BasicJsonType>::end_object() [with BasicJsonType = nlohmann::basic_json<>]'
json.hpp:8744:33:   required from 'bool nlohmann::detail::parser<BasicJsonType>::sax_parse_internal(SAX*) [with SAX = nlohmann::detail::json_sax_dom_callback_parser<nlohmann::basic_json<> >; BasicJsonType = nlohmann::basic_json<>]'
json.hpp:8635:31:   required from 'void nlohmann::detail::parser<BasicJsonType>::parse(bool, BasicJsonType&) [with BasicJsonType = nlohmann::basic_json<>]'
json.hpp:20732:79:   required from 'static nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer> nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer>::parse(IteratorType, IteratorType, nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer>::parser_callback_t, bool) [with IteratorType = const char*; typename std::enable_if<std::is_base_of<std::random_access_iterator_tag, typename std::iterator_traits<_InputIterator>::iterator_category>::value, int>::type <anonymous> = 0; ObjectType = std::map; ArrayType = std::vector; StringType = std::__cxx11::basic_string<char>; BooleanType = bool; NumberIntegerType = long int; NumberUnsignedType = long unsigned int; NumberFloatType = double; AllocatorType = std::allocator; JSONSerializer = nlohmann::adl_serializer; nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer>::parser_callback_t = std::function<bool(int, nlohmann::detail::parser<nlohmann::basic_json<> >::parse_event_t, nlohmann::basic_json<>&)>]'
json.hpp:22550:42:   required from here
/opt/wandbox/gcc-head/include/c++/10.0.0/bits/stl_uninitialized.h:135:72: error: 'value' is not a member of 'std::is_constructible<nlohmann::basic_json<>, const nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer>&>'
  135 |       static_assert(is_constructible<_ValueType2, decltype(*__first)>::value,
      |                                                                        ^~~~~
```

**Comment by melpon** [comment]
Fixed it and include path is also changed. You should be use `#include <nlohmann/json.hpp>`
